app/controllers/bookmarks_controller.rb

```diff
@@ -132,23 +132,27 @@ def status_action documents
     status = params['action']
     Array(documents.map(&:id)).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
+      sucess = false
+      case status
```

Comment:
I guess I disagree will this. What you propose works and uses fewer lines, but it assumes that those who can 'update' and those that can 'unpublish' are doing the same thing (true today, but might break in the future). I would rather have the unpublish activity tied to somebody who can 'unpublish', and the publish activity tied to somebody who can 'update' (would really have preferred it if it was tied to an ability called 'publish', but yeah I'm just working with what's there).

(Looking a few lines above this, I see I have assigned a variable called 'sucess'. Good thing nil is falsey ...)